File: app/src/main/java/huce/fit/appreadstories/adapters/ViewPagerStoryInterfaceAdapter.java
package huce.fit.appreadstories.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import huce.fit.appreadstories.fragment.StoryInterfaceRateFagment;
import huce.fit.appreadstories.fragment.StoryInterfaceIntroduceFagment;

public class ViewPagerStoryInterfaceAdapter extends FragmentStateAdapter {
    private final int idStory;

    public ViewPagerStoryInterfaceAdapter(@NonNull FragmentActivity fragmentActivity, int idStory) {
        super(fragmentActivity);
        this.idStory = idStory;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        if (position == 0) {
            return new StoryInterfaceIntroduceFagment(idStory);
        }
        return new StoryInterfaceRateFagment(idStory);
    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: app/src/main/java/huce/fit/appreadstories/adapters/ViewPagerStoryInterfaceAdapter.kt
package huce.fit.appreadstories.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import huce.fit.appreadstories.story.information.sub_fragment.introduce.IntroduceFragment
import huce.fit.appreadstories.story.information.sub_fragment.rate.RateFragment

class ViewPagerStoryInterfaceAdapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {

    override fun createFragment(position: Int): Fragment {
        return if (position == 0) {
            IntroduceFragment()
        } else {
            RateFragment()
        }
    }

    override fun getItemCount(): Int {
        return 2
    }
}
